app/controllers/users_controller.rb

```diff
@@ -1,40 +1,90 @@
 # frozen_string_literal: true
 
-# TODO: move this into a new UserController
-# see observer_controller.rb
-module ObserverController::UserController
+class UsersController < ApplicationController
+  # These need to be moved into the files where they are actually used.
+  require "find"
+  require "set"
+
+  before_action :login_required
+  before_action :disable_link_prefetching, except: [
+    :show,
+    :by_contribution
```

Comment:
Sorry to be picky, but show user is an expensive page.  I might prevent prefetching it, too.  Not sure about by_contribution.  Probably better to disable it, too.  Certainly, I don't expect it to be a common work flow!!

Reply:
Please be picky!

Reply:
Removed